api/pkg/notification/notification.go

```diff
@@ -0,0 +1,57 @@
+package notification
+
+import (
+	"fmt"
+
+	"github.com/wearedevx/keystone/api/internal/emailer"
+	"github.com/wearedevx/keystone/api/pkg/models"
+	"github.com/wearedevx/keystone/api/pkg/repo"
+)
+
+// Send email to amdins and users for every new devices created
+func SendEmailForNewDevices(r repo.IRepo) error {
+	var newDevices []models.Device
+	if err := r.GetNewlyCreatedDevices(&newDevices).Err(); err != nil {
+		return err
+	}
+
+	for _, device := range newDevices {
+		// Newly created devices only have one user
+		user := device.Users[0]
+
+		var adminProjectsMap map[string][]string
+		if err := r.GetAdminsFromUserProjects(user.ID, &adminProjectsMap).Err(); err != nil {
+			return err
+		}
+		for adminEmail, projectList := range adminProjectsMap {
+			// Send mail to admins of user projects
+			e, err := emailer.NewDeviceAdminMail(user.Username, projectList, device.Name)
+			if err != nil {
+				return err
+			}
+
+			if err = e.Send([]string{adminEmail}); err != nil {
+				fmt.Printf("Add New Device Admin Mail err: %+v\n", err)
+				return err
+			}
+		}
+
+		// Send mail to user
+		e, err := emailer.NewDeviceMail(device.Name, user.Username)
+		if err != nil {
+			return err
+		}
+
+		if user.Email != "" {
+			if err = e.Send([]string{user.Email}); err != nil {
+				fmt.Printf("Add New Device User Mail err: %+v\n", err)
+				return err
+			}
+		}
+		if err = r.GetDb().Model(&models.UserDevice{}).Where("user_id = ? and device_id = ?", user.ID, device.ID).Update("newly_created", false).Error; err != nil {
+			return err
+		}
```

Comment:
```suggestion
		if err = r.GetDb().Model(&models.UserDevice{}).Where("user_id = ? and device_id = ?", user.ID, device.ID).Update("newly_created", false).Error; err != nil {
			return err
		}
```

All interactions with the database should be done in the Repo, if there is a need for a new interaction, then there should be a new corresponding method to the Repo such as `func (r *Repo) SetNewlyCreatedDevice(flag bool) IRepo`

But also, if the notification is meant to wrap the emailer, why does it interact whith the database. I believe the database update should be donne by whoever calls this function instead.

Finally, the notification service is a good idea. There are other places where we send mail, maybe there is a way to put all those here, so that we have a uniform way of sending users notifications?